backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/payment/paymentallocation/service/PaymentAllocationService.java

```diff
@@ -0,0 +1,208 @@
+						.setParameter("paymentAllocationCriteria", paymentAllocationCriteria));
+
+		return builder.build();
+	}
+
+	@NonNull
+	private Optional<PaymentAllocationBuilder> preparePaymentAllocationBuilder(@NonNull final PaymentAllocationCriteria paymentAllocationCriteria)
+	{
+		final boolean paymentAllocationItemsMissing = CollectionUtils.isEmpty(paymentAllocationCriteria.getPaymentAllocationPayableItems());
+
+		if (paymentAllocationItemsMissing)
+		{
+			return Optional.empty();
+		}
+
+		final I_C_Payment payment = paymentAllocationCriteria.getPayment();
+
+		final PaymentDocument paymentDocument = toPaymentDocument(payment);
+
+		final ImmutableList<PayableDocument> invoiceDocuments = paymentAllocationCriteria.getPaymentAllocationPayableItems()
+				.stream()
+				.map(paymentAllocationPayableItem -> toPayableDocument(paymentAllocationPayableItem, paymentDocument))
+				.collect(ImmutableList.toImmutableList());
+
+		final LocalDate dateTrx = TimeUtil.asLocalDate(paymentAllocationCriteria.getDateTrx());
+		final PaymentAllocationBuilder builder = PaymentAllocationBuilder.newBuilder()
+				.invoiceProcessingServiceCompanyService(invoiceProcessingServiceCompanyService)
+				.dateTrx(dateTrx)
+				.dateAcct(dateTrx)
+				.paymentDocuments(ImmutableList.of(paymentDocument))
+				.payableDocuments(invoiceDocuments)
+				.allowPartialAllocations(paymentAllocationCriteria.isAllowPartialAllocations())
+				.payableRemainingOpenAmtPolicy(PaymentAllocationBuilder.PayableRemainingOpenAmtPolicy.DO_NOTHING)
+				.allowPurchaseSalesInvoiceCompensation(paymentAllocationBL.isPurchaseSalesInvoiceCompensationAllowed());
+
+		return Optional.of(builder);
+	}
+
+	@VisibleForTesting
+	PaymentDocument toPaymentDocument(@NonNull final I_C_Payment payment)
+	{
+		final PaymentDirection paymentDirection = extractPaymentDirection(payment);
+
+		final Money openAmt = extractPayAmt(payment).negateIf(paymentDirection.isOutboundPayment());;
+
+		return PaymentDocument.builder()
+				.paymentId(PaymentId.ofRepoId(payment.getC_Payment_ID()))
+				.bpartnerId(BPartnerId.ofRepoId(payment.getC_BPartner_ID()))
+				.documentNo(payment.getDocumentNo())
+				.paymentDirection(paymentDirection)
+				.openAmt(openAmt)
+				.amountToAllocate(openAmt)
+				.dateTrx(TimeUtil.asLocalDate(payment.getDateTrx()))
+				.clientAndOrgId(ClientAndOrgId.ofClientAndOrg(payment.getAD_Client_ID(), payment.getAD_Org_ID()))
+				.currencyConversionTypeId(CurrencyConversionTypeId.ofRepoIdOrNull(payment.getC_ConversionType_ID())) // ??
+				.build();
+	}
+
+	private PaymentDirection extractPaymentDirection(final I_C_Payment payment)
+	{
+		return PaymentDirection.ofReceiptFlag(payment.isReceipt());
+	}
+
+	private Money extractPayAmt(@NonNull final I_C_Payment payment)
+	{
+		final CurrencyId currencyId = CurrencyId.ofRepoId(payment.getC_Currency_ID());
+		return Money.of(payment.getPayAmt(), currencyId);
+	}
+
+	@VisibleForTesting
+	PayableDocument toPayableDocument(
```

Comment:
this is a duplication of de.metas.ui.web.payment_allocation.process.PaymentsViewAllocateCommand#toPayableDocument.